tools/crm_mon.c

```diff
@@ -4418,7 +4553,7 @@ clean_up(crm_exit_t exit_code)
             fprintf(stdout, "Content-Type: text/plain\n"
                             "Status: 500\n\n");
         } else {
-            crm_help('?', CRM_EX_USAGE);
+            fprintf(stderr, "%s", g_option_context_get_help(context, TRUE, NULL));
```

Comment:
crm_help() also exits. In this case setting exit_code to CRM_EX_USAGE would handle it.

Reply:
As I've been converting programs to use glib for command line handling, I have stopped using crm_help in them because it doesn't understand how to construct output from a GOptionContext.  I figured eventually that function (and the other command line handling functions) would just go away.

Reply:
Definitely. But we need to exit CRM_EX_USAGE

Reply:
Right, so maybe I am missing something but the only way that fprintf is going to get called in the first place is if exit_code is CRM_EX_USAGE, and then there's a crm_exit right after it.  Perhaps that isn't clear from looking at the diff, though.

Reply:
Ah, sorry, I didn't look at the context